src/snowflake/snowpark/_internal/ast_utils.py

```diff
@@ -346,6 +346,28 @@ def build_expr_from_snowpark_column_or_python_val(
         pass
     else:
         raise TypeError(f"{type(value)} is not a valid type for Column or literal AST.")
+    
+
+def build_expr_from_snowpark_column_or_table_fn(
+    expr_builder: proto.Expr, value: Union["snowflake.snowpark.Column", "snowflake.snowpark.table_function.TableFunctionCall"]
+) -> None:
+    """Copy from a Column object's AST, or TableFunctionCall object's AST, into an AST expression.
+
+    Args:
+        expr_builder (proto.Expr): A previously created Expr() or SpColumnExpr() IR entity intance to be filled
```

Comment:
should the type hint be `Union[proto.Expr, proto.SpColumnExpr]`?

Reply:
Actually the description of the argument is incorrect here, my mistake. I'm noticing this error has propagated down across the helper methods so I'll clean that up.

Just to be clear I think we moved to represent `Column` instances within `Expr` IR entities to accommodate both `SpColumnExpr` variants and values from `const.ir`. So all `expr_builder` arguments should be only `proto.Expr` instances.